enterprise/migrations/0218_add_provisioning_admin_role.py

```diff
@@ -0,0 +1,38 @@
+# Generated by Django 4.2.13 on 2024-07-24 13:03
+
+from django.db import migrations
+
+
+from enterprise.constants import SYSTEM_ENTERPRISE_PROVISIONING_ADMIN_ROLE
+
+
+def create_enterprise_provisioning_admin_role(apps, schema_editor):  # pylint: disable=unused-argument
+    """Create the `enterprise_provisioining_admin` system-wide role if it does not already exist"""
+    SystemWideEnterpriseRole = apps.get_model(
+        'enterprise', 'SystemWideEnterpriseRole')
+    SystemWideEnterpriseRole.objects.update_or_create(
+        name=SYSTEM_ENTERPRISE_PROVISIONING_ADMIN_ROLE,
+        description='Role for provisioning admins',
+    )
+
+
+def delete_enterprise_provisioning_admin_role(apps, schema_editor):  # pylint: disable=unused-argument
+    """Delete the `enterprise_provisioining_admin` system-wide role"""
+    SystemWideEnterpriseRole = apps.get_model(
+        'enterprise', 'SystemWideEnterpriseRole')
+    SystemWideEnterpriseRole.objects.filter(
```

Comment:
Any reason we are deleting the role after creating it?

Reply:
it runs in case of roll back only.